### PR TITLE
Implement depth measurements from lidar camera

### LINKED ISSUE #42: [Pod] Implement depth measurements from lidar camera
To have a basic collision detection system and/or end-goal identification, we'll be using a LIDAR camera to obtain a depth map of the visual area in front of the pod.
- Create an abstraction for the lidar camera
  - Establish serial communication over USB with the camera
- Provide a method to acquire the depth map matrix from the camera
  - Refer to the [OnionIoT/tau-lidar-camera](OnionIoT/tau-lidar-camera) implementation for how to read and process data frames

### PR DESCRIPTION
**Pull request by ryescholin**
not using rppal bc of lack of support -- tested w sam
Create `Lidar` abstraction to measure distance of object in line of sight.
Resolves #42.

### CODE REVIEW

**Review by SM-dot — APPROVED** [review]

File: pod-operation/src/components/lidar.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+pub struct Lidar {
+	lidar_lite: LidarLiteV3<LinuxI2CDevice>,
+}
+
+impl Lidar {
+	pub fn new() -> Lidar {
+		// Initialize the I2C device
+		let i2cdev =
+			LinuxI2CDevice::new("/dev/i2c-1", 0x62).expect("Failed to initialize I2C device");
```
Comment (taesungh) [inline pod-operation/src/components/lidar.rs:12]: Chore: could we extract the I2C address to a named global constant

File: pod-operation/src/components/lidar.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+		// Create LidarLiteV3 instance
+		let lidar_lite = LidarLiteV3::new(i2cdev).expect("Failed to initialize LidarLiteV3");
+
+		Lidar { lidar_lite }
+	}
+
+	pub fn read_distance(&mut self) -> f32 {
+		let distance = self.lidar_lite.read_distance(false).unwrap();
+		distance as f32 / 100.0
```
Comment (taesungh) [inline pod-operation/src/components/lidar.rs:21]: Chore: casting with `as` should generally be avoided for possibilities of lossy conversion. Consider using `f32::from` instead.

**Review by taesungh — COMMENTED** [review]
Unfortunate there's no generic `embedded-hal` support. Perhaps we could eventually look into reworking the library implementation, but shouldn't be a huge concern now. Some minor notes about numeric conversions and improving clarity.

**Review by taesungh — APPROVED** [review]
Thanks for adding, trusting the device was tested and works.

**Pull request merged**